### PR TITLE
Port the feature to support no inline function call from Triton.

### PR DESCRIPTION
**Pull request by chengjunlu**

### CODE REVIEW

**Review by Stonepia — APPROVED** [review]
LGTM, this PR can help with the tests of `test_noinline`

**Review by Stonepia — APPROVED** [review]
LGTM, this PR can help with the tests of `test_noinline`

File: lib/Conversion/TritonGPUToSPIRV/TritonGPUToSPIRVPass.cpp [hunk 1]
```diff
@@ -139,17 +181,17 @@ struct FuncOpConversion : public FuncOpConversionBase {
...
+      //    // for `nvvm.annotation` metadata.
+      //    newFuncOp->setAttr("nvvm.maxntid", rewriter.getI32ArrayAttr(32 * numWarps));
       newFuncOp->setAttr(spirv::getEntryPointABIAttrName(),
                          spirv::EntryPointABIAttr::get(getContext(), nullptr, std::nullopt));
     } else {
       // The noinline function control will be used by the SPIRV codegen to prevent
       // inlining.
       newFuncOp.setFunctionControl(spirv::FunctionControl::DontInline);
-//      rewriter.eraseOp(amendedFuncOp);
+      rewriter.eraseOp(amendedFuncOp);
```
[the remark this answers is not in the document]
Comment (chengjunlu) [reply lib/Conversion/TritonGPUToSPIRV/TritonGPUToSPIRVPass.cpp:193]: To follow the new logic. We temply create a amendedFuncOp in `tt.func` and then convert it to `spriv.func`. Here we need to remove the temply created amendedFuncOp .

File: lib/Conversion/TritonGPUToSPIRV/TritonGPUToSPIRVPass.cpp [hunk 2]
```diff
@@ -139,17 +181,17 @@ struct FuncOpConversion : public FuncOpConversionBase {
 
     if (allocation.isRoot(funcOp)) {
       // Set an attribute to indicate this function is a kernel entry.
+      //    // Set an attribute for maxntidx, it could be used in latter LLVM codegen
+      //    // for `nvvm.annotation` metadata.
+      //    newFuncOp->setAttr("nvvm.maxntid", rewriter.getI32ArrayAttr(32 * numWarps));
```
[the remark this answers is not in the document]
Comment (chengjunlu) [reply lib/Conversion/TritonGPUToSPIRV/TritonGPUToSPIRVPass.cpp:186]: No. But here maybe a hint for me to add the sub-group size into the function.

**Review by EikanWang — APPROVED** [review]

**Pull request merged**